src/Mod/Draft/draftviewproviders/view_style_dimension.py

```diff
@@ -0,0 +1,230 @@
+        """
+        Initializes Dimension Style properties
+        """
+        # get the style from FreeCAD Draft Parameters
+        param = App.ParamGet("User parameter:BaseApp/Preferences/Mod/Draft")
+        annotation_scale = param.GetFloat("DraftAnnotationScale", 1.0)
+
+        vobj.ScaleMultiplier = 1 / annotation_scale
+        vobj.AutoUpdate = True
+
+        vobj.FontName = utils.get_param("textfont","")
+        vobj.FontSize = utils.get_param("textheight",0.20)
+        vobj.TextSpacing = utils.get_param("dimspacing",0.05)
+
+        vobj.Decimals = utils.get_param("dimPrecision",2)
+        vobj.ShowUnit = utils.get_param("showUnit",True)
+
+        vobj.ArrowSize = utils.get_param("arrowsize",0.1)
+        vobj.ArrowType = utils.ARROW_TYPES
+        vobj.ArrowType = utils.ARROW_TYPES[utils.get_param("dimsymbol",0)]
+        vobj.DimOvershoot = utils.get_param("dimovershoot",0)  
+        vobj.ExtLines = utils.get_param("extlines",0.3)
+        vobj.ExtOvershoot = utils.get_param("extovershoot",0)
+        vobj.ShowLine = True          
+
+        if existing_dimension and hasattr(existing_dimension, "ViewObject"):
+            # get the style from given dimension
+            from draftutils import gui_utils
+            gui_utils.format_object(target = vobj.Object, origin = existing_dimension)
+          
+    def onChanged(self, vobj, prop):
+        if hasattr(vobj, "AutoUpdate"):
+            if vobj.AutoUpdate:
+                self.update_related_dimensions(vobj)
+
+    def doubleClicked(self,vobj):
+        self.set_current(vobj)
+
+    def setupContextMenu(self,vobj,menu):
+        action1 = menu.addAction("Set current")
+        action1.triggered.connect(lambda f=self.set_current, arg=vobj:f(arg))
+        action2 = menu.addAction("Update dimensions")
+        action2.triggered.connect(lambda f=self.update_related_dimensions, arg=vobj:f(arg))
+    
+    def set_current(self, vobj):
+        """
+        Sets the current dimension style as default for new created dimensions
+        """
+        param = App.ParamGet("User parameter:BaseApp/Preferences/Mod/Draft")
+        param.SetFloat("DraftAnnotationScale", 1 / vobj.ScaleMultiplier)
+
+        param.SetString("textfont", vobj.FontName)
+        param.SetFloat("textheight", vobj.FontSize)
+        param.SetFloat("dimspacing", vobj.TextSpacing)
+ 
+        param.SetInt("dimPrecision", vobj.Decimals)
+
+        param.SetFloat("arrowsize", vobj.ArrowSize)
+        param.SetInt("dimsymbol", utils.ARROW_TYPES.index(vobj.ArrowType))
+        param.SetFloat("dimovershoot", vobj.DimOvershoot)
+        param.SetFloat("extlines", vobj.ExtLines)
+        param.SetFloat("extovershoot", vobj.ExtOvershoot)
+        
+        App.Console.PrintMessage("Current dimension style set to " + str(vobj.Object.Label) + "\n")
+
+    def update_related_dimensions(self, vobj):
+        """
+        Apply the style to the related dimensions
+        """
+        from draftutils import gui_utils
```

Comment:
This import should be at the top of the file, after `draftutils.utils`, not inside the function.